### ISSUE TITLE
Issues with demos in IE 10

### ISSUE THREAD

**Issue by alexander-sheehan** [GoogleChrome/dialog-polyfill#147]
I'm trying to run the demos provided at https://demo.agektmr.com/dialog/ in IE 10 and am getting the prompt: "Your browser doesn't seem to support dialog element yet. This demo is running in polyfill mode." However, none of the demo's seem to be functional. Buttons that are supposed to be on the dialog box are being displayed by default and all buttons don't seem to have any function.

**Comment by samthor** [comment]
Is there any interesting console etc output? The demo page should be inserting the relevant script and CSS (can you check actually? The Angular code in [main.js](https://demo.agektmr.com/dialog/js/main.js) should do it—although I do wonder whether Angular supports IE10.

We can test in IE10/9, but it's a struggle.

**Comment by alexander-sheehan** [comment]
<img width="656" alt="screen shot 2017-07-07 at 11 20 22 am" src="https://user-images.githubusercontent.com/29286674/27971164-71c6c3de-6306-11e7-8a3b-993b4b54be94.png"> 

This is the console output upon loading the page.

<img width="846" alt="screen shot 2017-07-07 at 11 20 49 am" src="https://user-images.githubusercontent.com/29286674/27971162-71ba09fa-6306-11e7-9cad-4d4735ef82df.png">

<img width="1253" alt="screen shot 2017-07-07 at 11 19 38 am" src="https://user-images.githubusercontent.com/29286674/27971163-71bdbe88-6306-11e7-971f-bf40f53df7f2.png">

These are two screen shots of two particular demos on the page, but all of demos are failing

**Comment by samthor** [comment]
This is a problem with the demo site, not with `dialog`. We probably won't fix it, but if it's important I can come up with a simpler one.

If you have any problems with using `dialog` on IE9/10 please file new bugs.

**Comment by NaggingDaivy** [comment]
Hello,
I am using this polyfill in my application and it does not work with IE9 or IE10, but it works with IE11. I am getting the same errors as in the demos.
So, I suppose that support for IE9 and IE10 is not available?

**Comment by sntran** [comment]
This is the error I see in IE10:

"In strict mode, function declarations cannot be nested inside a statement or block. They may only appear at the top level or directly inside a function body. "

This comes from this code block:

```js
    var nativeFormSubmit = HTMLFormElement.prototype.submit;
    function replacementFormSubmit() {
      if (!isFormMethodDialog(this)) {
        return nativeFormSubmit.call(this);
      }
      var dialog = findNearestDialog(this);
      dialog && dialog.close();
    }
```

This is fixed on master (through PR #160), but has not been published to NPM yet.

**Comment by samthor** [comment]
I've published 0.4.10 to fix this.

**Comment by samthor** [comment]
I've published new demos however they use modern JS/CSS and probably won't work in IE10.
